src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Functions/Processor.cs

```diff
@@ -165,6 +123,134 @@ await _validatorService.Validate(message.Release.Id, subjectData, executionConte
             }
         }
 
+        private async Task ProcessUnpackingArchive(
+            ImportMessage message,
+            ILogger logger)
+        {
+            if (message.ArchiveFileName != "")
+            {
+                logger.LogInformation($"Unpacking archive for {message.DataFileName}");
+
+                var status =
+                    await _importStatusService.GetImportStatus(message.Release.Id, message.OrigDataFileName);
+
+                if (status.IsAfterArchiveProcessing())
+                {
+                    logger.LogInformation($"Unpacking archive already completed for {message.DataFileName} - skipping");
+                    return;
+                }
+                
+                await _importStatusService.UpdateStatus(message.Release.Id,
+                    message.DataFileName,
+                    IStatus.PROCESSING_ARCHIVE_FILE);
+
+                await _dataArchiveService.ExtractDataFiles(message.Release.Id, message.ArchiveFileName);
+            }
+        }
+
+        private async Task ProcessStage1(
+            ImportMessage message,
+            ILogger logger,
+            ExecutionContext executionContext,
+            SubjectData subjectData)
+        {
+            logger.LogInformation($"Processing Stage 1 for {message.DataFileName}");
```

Comment:
Same for these other stage logs in the rest of this file. Would move them to after the stage completed check to avoid any confusion and reduce log noise.

Reply:
Agreed and done